### PR TITLE
[Build] Workflow to generate unidoc

### LINKED ISSUE #2209: [Feature Request][Kernel] Run unidoc for all changes
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [x] Kernel
- [ ] Other (fill in here)

### Overview
Currently the Kernel or Connector CI job just runs the tests and doesn't generate any docs to make sure there are no errors introduced in the PRs. Doc generation is [done as part of the `delta-spark` tests](https://github.com/delta-io/delta/blob/master/run-tests.py#L72) which are not run if the PR just contains Kernel only changes. 

May be it is better to separate out the unidoc generation into a separate CI workflow that always runs?

### PR DESCRIPTION
**Pull request by sokolat**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [x] Kernel
- [ ] Other (Doc)

## Description

Resolves #2209 
this PR will make sure all the component apis are documented.
Documentation will be generated for all changes
## How was this patch tested?
commit triggered workflow tests

### CODE REVIEW

File: .github/workflows/unidoc.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+  name: "Unidoc generation"
```
Comment (vkorukanti) [inline .github/workflows/unidoc.yaml:1]: This is great. Given that now we have this job, can you remove the doc generation from [here](https://github.com/delta-io/delta/blob/master/run-tests.py#L73)?

File: .github/workflows/unidoc.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      name: "Generate unidoc"
+      runs-on: ubuntu-20.04
+      strategy:
+        matrix:
+        # These Scala versions must match those in the build.sbt
+          scala: [2.13.8, 2.12.17]
+      steps:
+        - uses: actions/checkout@v3
+        - name: generate unidoc
+          run: build/sbt "++ ${{ matrix.scala }}" unidoc
```
Comment (vkorukanti) [inline .github/workflows/unidoc.yaml:14]: nit: add a new line EOF

**Review by vkorukanti — CHANGES_REQUESTED** [review]
This is great! thank you for the PR.

Looks like the workflow is failing currently. Could you please check that is because we have a failure on the master branch?

**Comment by sokolat** [comment]
> This is great! thank you for the PR.
> 
> Looks like the workflow is failing currently. Could you please check that is because we have a failure on the master branch?

The unidoc generation workflow is failing in master branch because of package import issues. It's not able to find the packaging structure. Is it related to some configuration in `build.sbt` file ?

File: .github/workflows/unidoc.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+  name: "Unidoc generation"
+  on: [pull_request]
```
Comment (allisonport-db) [inline .github/workflows/unidoc.yaml:2]: @vkorukanti Do we want it to also be on push? I think all our other jobs are

Comment (vkorukanti) [reply .github/workflows/unidoc.yaml:2]: yep, we should. Updating..

**Review by allisonport-db — APPROVED** [review]
LGTM

**Review by tdas — APPROVED** [review]

**Pull request merged**